patches/minecraft/net/minecraft/item/Item.java.patch

```diff
@@ -714,6 +714,32 @@
 +        }
 +        return builder.build();
 +    }
++    
++    /**
++     * @param stack - The armor stack
++     * @return True if the itemstack is valid horse armor, false if it isn't.
++     */
++    public boolean isValidHorseArmor(ItemStack stack)
++    {
++        return stack.func_77973_b() == Items.field_151138_bX || stack.func_77973_b() == Items.field_151136_bY || stack.func_77973_b() == Items.field_151125_bZ;
++    }
++    
++    /** 
++     * @param stack - The armor stack
++     * @return The HorseArmorType of the armor stack 
++     */
++    @Nullable
++    public net.minecraft.entity.passive.HorseArmorType getHorseArmorType(ItemStack stack)
++    {
++        return stack.func_77973_b() == Items.field_151138_bX ? net.minecraft.entity.passive.HorseArmorType.IRON : stack.func_77973_b() == Items.field_151136_bY ? net.minecraft.entity.passive.HorseArmorType.GOLD : stack.func_77973_b() == Items.field_151125_bZ ? net.minecraft.entity.passive.HorseArmorType.DIAMOND : null;
++    }
++    
++    /**
++     * @param world - The world the horse is in
++     * @param horse - The horse wearing the armor
++     * @param itemStack - the armor itemstack
++     */
++    public void onHorseArmorTick(World world, net.minecraft.entity.passive.EntityHorse horse, ItemStack itemStack) {}
```

Comment:
As far as I'm aware, Forge discourages that. I think this did change with 1.12 though, can you confirm that?